Phase03/FullTextSearch/Controllers/Logic/MathHelper.cs

```diff
@@ -0,0 +1,36 @@
+namespace FullTextSearch.Controllers.Logic;
+
+public static class MathHelper
+{
+    public static List<T> Union<T>(this List<List<T>> enumberList)
+    {
+        try
+        {
+            if (!enumberList.Any())
+                return new List<T>();
+
+            return enumberList.Aggregate((current, next) => current.Union(next).ToList());
+        }
+        catch (ArgumentNullException e)
+        {
+            Console.WriteLine(e);
+            throw;
+        }
```

Comment:
never catch system exceptions
you can define custom exception and handle it if you want